### ISSUE TITLE
SigNoz Integration

### ISSUE THREAD

**Issue by david1542** [merlinn-co/merlinn#26]
Add a basic integration with SigNoz. The integration, in the initial phase, should do the following:
* Read alerts information via the API
* Read logs via the API

**Comment by nitin302** [comment]
Logs API - https://signoz.io/docs/logs-management/logs-api/overview/#api-endpoint
Alerts API - maybe instead of Signoz directly, we can integrate with one of the integrations it has like pagerduty?

**Comment by nitin302** [comment]
Also, for Logs API since community edition does not have apikey support, if we can get support via user credentials that would make integration more widely used.

**Comment by david1542** [comment]
Thanks for sharing the references to SigNoz docs :) @nitin302 
About the alerts API - since we're already integrated with PagerDuty/Opsgenie, it'd indeed be simpler to stick with that. Namely, rely on SigNoz integration with them and integrate with PD/Opsgenie.

About the logs API - can you please elaborate on the API key v.s basic credentials? Are you referring to SigNoz community edition?

BTW - If you want to take a stab at that and implement and initial version of SigNoz integration, that would be amazing :) I can help you on that.

**Comment by nitin302** [comment]
Yes, the SigNoz community edition does not have API key support - https://signoz.io/pricing/, but typically user credentials (basic) works.

I might be interested in contributing, if you can share steps... I can create a new PR.

**Comment by david1542** [comment]
@nitin302 sure, I can provide some guidance :) Thanks for showing interest.

For the SigNoz integration, I recommend following the structure we used for Coralogix:

**Database**: Add a new record in the vendors table by updating `packages/db/src/models/vendor.ts` and including SigNoz in the vendorsData array.

**UI Card**: Copy the Coralogix UI card in `services/dashboard/src/components/Connection/integrations/basic`, adjust it for SigNoz, and configure the necessary fields like username, password, and host.

**Sub-Agent**: Create a sub-agent in Langchain based on the Coralogix structure, modifying it for SigNoz. Currently, the Coralogix agent fetches logs, so we’ll need to decide what SigNoz should do.

**(Optional) Processor**: If needed, create a processor for Opsgenie incidents to handle SigNoz-related root-cause-analysis flows.

The main work is to create the sub-agent. This is the trickiest part, and requires some prompt engineering.
The agent needs to understand the underlying topology, log structures, service names, etc. We did some work on Coralogix and now the agent is in a pretty decent place, but it's still not perfect.

Lemme know what you think about it. I'd be happy to hop on a call with you if you want to discuss further :)

**Comment by vikash-fourkites** [comment]
@david1542 @nitin302 I am going to attempt Signoz integration. We will see how it turns out.

**Comment by david1542** [comment]
@vikash-fourkites Gotcha, thanks for good spirit of contributing 😊

I know @nitin302 has started working on that. @nitin302 Do you mind sharing your progress maybe?

**Comment by nitin302** [comment]
I wasn't able to make much progress but i've provided context to @vikash-fourkites, thanks.

**Comment by david1542** [comment]
@nitin302 awesome, thanks a lot.
@vikash-fourkites if you need any help, feel free to reach out to me here or on [Slack](https://join.slack.com/t/merlinncommunity/shared_invite/zt-2le6bbavn-1WMwsPU5noECPyV9nQ8h4g) :)

**Comment by Subash-Mohan** [comment]
Hi @david1542, I noticed there haven't been any updates on the issue since September 18. I'm interested in contributing and would love to take it on if possible. Could you also send me an invitation to join the Slack? Thanks!

**Comment by david1542** [comment]
Hi @Subash-Mohan :) Thanks for reaching out! We're currently working on it ourselves. You can join the Slack community using this [link](https://join.slack.com/t/vesppercommunity/shared_invite/zt-2le6bbavn-1WMwsPU5noECPyV9nQ8h4g).
If you want, there are some other issues that you can take.
